# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jsapi.cpp
# Commit: e862899dca3f
# Full Hash: e862899dca3f252a8fe3c7be10d83d79dad328c5
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2018-05-16 18:32:17
# Regressor Bug: 1461751
# File Overlap Count: 2
# Description:
#   Bug 1461751 - Simplify module resolve hook to be a function pointer r=luke r=baku
# ==============================================================================

diff -r 1076e77e7b5c -r e862899dca3f js/src/jsapi.cpp
--- a/js/src/jsapi.cpp	Tue Apr 24 16:19:51 2018 +0200
+++ b/js/src/jsapi.cpp	Wed May 16 11:59:09 2018 +0100
@@ -4939,21 +4939,18 @@
     return ::Evaluate(cx, optionsArg, filename, rval);
 }
 
-JS_PUBLIC_API(JSFunction*)
-JS::GetModuleResolveHook(JSContext* cx)
-{
-    AssertHeapIsIdle();
-    CHECK_REQUEST(cx);
-    return cx->global()->moduleResolveHook();
+JS_PUBLIC_API(JS::ModuleResolveHook)
+JS::GetModuleResolveHook(JSRuntime* rt)
+{
+    AssertHeapIsIdle();
+    return rt->moduleResolveHook;
 }
 
 JS_PUBLIC_API(void)
-JS::SetModuleResolveHook(JSContext* cx, HandleFunction func)
-{
-    AssertHeapIsIdle();
-    CHECK_REQUEST(cx);
-    assertSameCompartment(cx, func);
-    cx->global()->setModuleResolveHook(func);
+JS::SetModuleResolveHook(JSRuntime* rt, JS::ModuleResolveHook func)
+{
+    AssertHeapIsIdle();
+    rt->moduleResolveHook = func;
 }
 
 JS_PUBLIC_API(bool)